File: Baekjoon/Gold/2467_용액/2467.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const N = parseInt(input[0]);
const arr = input[1].split(" ").map(Number);

let left = 0;
let right = N - 1;
let min_value = Infinity;
let result = [];

while (left < right) {
    let mix = arr[left] + arr[right]
    if (abs(mix) < min_value ) {
        min_value = abs(mix)
        result = [arr[left], arr[right]]
    }

    if (mix > 0) {
        right --
    } else {
        left ++
    }
}

console.log(result[0], result[1]);

File: Baekjoon/Gold/2467_용액/2467.py
N = int(input()) # 전체 용액의 수
arr = list(map(int, input().split()))

min_value = float("inf")
min_list = []
current_value = 0

i = 0
j = N-1
while i <= (N//2+1) and j >= (N//2-1) and arr[i] != arr[j]:
    current_value = arr[i] + arr[j]
    if min_value > abs(current_value):
        min_value = abs(current_value)
        min_list = [arr[i], arr[j]]
    else:
        break
    if 0 < current_value:
        j -= 1
    elif 0 > current_value:
        i += 1
        

print(min_list)
